apis/management.cattle.io/v3/logging_types.go

```diff
@@ -63,8 +63,9 @@ type ProjectLoggingSpec struct {
 }
 
 type ClusterLoggingStatus struct {
-	Conditions  []LoggingCondition `json:"conditions,omitempty"`
-	AppliedSpec ClusterLoggingSpec `json:"appliedSpec,omitempty"`
+	Conditions  []LoggingCondition  `json:"conditions,omitempty"`
+	AppliedSpec ClusterLoggingSpec  `json:"appliedSpec,omitempty"`
+	FailedSpec  *ClusterLoggingSpec `json:"failedSpec,omitempty"`
```

Comment:
This field doesn't seem to be used in corresponding Rancher PR; can we remove it for now?

Reply:
FailedSpec and AppliedSpec are used to prevent the infinite loop, the condition update will also trigger the sync function, once the AppliedSpec equal to the Spec, will not continue. The FailedSpec is for store the spec apply failed, if the same spec applies failed, will back off, and sync after the backoff time.